gdk/commands/component/PublishCommand.py

```diff
@@ -119,14 +118,24 @@ def _update_region(self):
 
     def _update_component_version(self):
         self.project_config["component_version"] = self.get_component_version_from_config()
+        self._update_publish_recipe_file_path()
+
+    def _update_publish_recipe_file_path(self):
+        ext = self.project_config["component_recipe_file"].name.split(".")[-1]
+        publish_recipe_file_name = "{}-{}.{}".format(
+            self.project_config["component_name"], self.project_config["component_version"], ext
+        )  # Eg. HelloWorld-1.0.0.yaml
+        self.project_config["publish_recipe_file"] = (
```

Comment:
Is there a way we could not mutate the project_config? Mutating this is specially bad given we treat this guy as a god object that gets passed around. If not lets add a  TODO, to avoid doing this